File: TypeScript/128bit/Rijandel.ts
import * as crypto from 'crypto';

function aesEncrypt(plaintext: string, key: Buffer): string {
    const iv = crypto.randomBytes(16);
    const cipher = crypto.createCipheriv('aes-192-cbc', key, iv);
    let encrypted = cipher.update(plaintext, 'utf8', 'hex');
    encrypted += cipher.final('hex');
    return iv.toString('hex') + encrypted;
}

function main() {
    const key = crypto.randomBytes(24); // 192-bit key
    const plaintext = "This is a test message for AES-192 encryption!";
    
    const ciphertext = aesEncrypt(plaintext, key);
    
    console.log("Ciphertext is: " + ciphertext);
}

main();

File: Ruby/128bit/Rijandel.rb
require 'openssl'
require 'securerandom'

def aes_encrypt(plaintext, key)
  cipher = OpenSSL::Cipher.new('aes-128-cbc')
  cipher.encrypt
  iv = cipher.random_iv
  cipher.key = key

  ciphertext = cipher.update(plaintext) + cipher.final
  iv + ciphertext
end

def main
  key = SecureRandom.random_bytes(16) # 128-bit key
  plaintext = "This is a test message for AES-128 encryption!"

  ciphertext = aes_encrypt(plaintext, key)

  puts "Ciphertext is: #{ciphertext.unpack1('H*')}"
end

main
